src/Stackage/Database/Cron.hs

```diff
@@ -218,10 +218,11 @@ runStackageUpdate doNotUpload = do
     runStackageMigrations
     didUpdate <- forceUpdateHackageIndex (Just "stackage-server cron job")
     case didUpdate of
-        UpdateOccurred -> do
```

Comment:
I'm not sure that this change is necessary. New deprecated information can only be available if there was an update to the Hackage index AFAIK.

Reply:
Yeah, that's what I previously though too. When testing the fix I saw that deprecated.json had changed, but `stackage-server-cron` wasn't picking it up. I just did it once again to confirm it:

Here is the info right before I undeprecate `massiv-scheduler`:

```shell
$ curl -s https://hackage.haskell.org/packages/deprecated.json | jq '.[] | select(."deprecated-package" == "massiv-scheduler")'
{
  "deprecated-package": "massiv-scheduler",
  "in-favour-of": [
    "scheduler"
  ]
}
$ curl -is https://hackage.haskell.org/01-index.tar.gz
HTTP/1.1 200 OK
Server: nginx/1.14.0 (Ubuntu)
Content-Type: application/x-gzip
Accept-Ranges: bytes
Cache-Control: public, no-transform, max-age=300
Content-MD5: Gep3JQkByU6ANrlE/ku1bw==
ETag: "19ea77250901c94e8036b944fe4bb56f"
Last-modified: Tue, 30 Jul 2019 05:53:51 GMT
Content-Length: 81877665
Accept-Ranges: bytes
Date: Tue, 30 Jul 2019 08:34:51 GMT
Via: 1.1 varnish
Connection: keep-alive
X-Served-By: cache-fra19176-FRA
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1564475691.307618,VS0,VE314
```

Here I gave it some time just to see if `01-index.tar.gz` would change, luckily noone uploaded any packages in the meantime. Note I also added an extra argument `x=2`, just to make sure I am not seeing a cached version. So the `massiv-scheduler` was undeprecated, but the index file is still the same, which makes sense, since deprecation info isn't part of the index file.

```shell
$ curl -s https://hackage.haskell.org/packages/deprecated.json?x=2 | jq '.[] | select(."deprecated-package" == "massiv-scheduler")'
$ curl -is https://hackage.haskell.org/01-index.tar.gz?x=2
HTTP/1.1 200 OK
Server: nginx/1.14.0 (Ubuntu)
Content-Type: application/x-gzip
Accept-Ranges: bytes
Cache-Control: public, no-transform, max-age=300
Content-MD5: Gep3JQkByU6ANrlE/ku1bw==
ETag: "19ea77250901c94e8036b944fe4bb56f"
Last-modified: Tue, 30 Jul 2019 05:53:51 GMT
Content-Length: 81877665
Accept-Ranges: bytes
Date: Tue, 30 Jul 2019 08:44:37 GMT
Via: 1.1 varnish
Connection: keep-alive
X-Served-By: cache-fra19133-FRA
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1564476277.957349,VS0,VE786
```